### ISSUE TITLE
Support type 1 AppImages

### ISSUE THREAD

**Issue by TheAssassin** [TheAssassin/AppImageLauncher#3]
Type 1 AppImages are a legacy technology. Unfortunately, a lot of people still rely on the deprecated and discontinued tools to build such AppImages, because they think they need some minor features like the thumbnailer (although that should've been resolved thanks to @azubieta -- TODO: verify this).

Therefore, AppImageLauncher should support them. This involves developing another "how to run AppImages without triggering AppImageLauncher again" strategy, which was quite a challenge to solve for type 2 already.

Definitely not a priority at the beginning phase. I consider this a "nice to have" feature. Priority might change in the future.

**Comment by azubieta** [comment]
Thumbnail generation for type 1 appimage works but there are some cases (not documented yet) in which it fails.

**Comment by TheAssassin** [comment]
> Thumbnail generation for type 1 AppImages works but there are some cases (not documented yet) in which it fails.

I was talking about the thumbnailer that is planned to be built into the runtime.

**Comment by TheAssassin** [comment]
I'll try to extract the runtime from type 1 AppImages into a separate file, patch away the magic bytes, and check whether by calling that runtime with a fake `argv[0]` pointing to the original AppImage's path the AppImage will run despite the binfmt set up.

**Comment by TheAssassin** [comment]
The method suggested in my previous comment didn't work very well for various reasons. First of all, Qt's `QProcess` can't be used for the purpose, as it doesn't allow overriding of `argv[0]`. Also it failed to start any kind of process I tried. I probably misused it. Anyway, we'd need some manual process management, which seems a little overkill for the purpose.

My favorite solution would be to have something like with type 2, a "precompiled" runtime that has an option to run an external AppImage. There is no such functionality in the old type 1 master branch of AppImageKit, though. Therefore, until I get to implementing this, I implemented the quick fix for the issue that I never wanted to have: copy the file, nuke the magic bytes, and run the AppImage from a temporary directory.

After finishing my testing, I'll push some commits implementing that workflow. We should search for an alternative, though. The method is _really_ inefficient.

**Comment by probonopd** [comment]
> Type 1 AppImages are a legacy technology. Unfortunately, a lot of people still rely on the deprecated and discontinued tools to build such AppImages

No, Type 1 is not deprecated and not legacy. Some types of users have valid reasons to use Type 1. Different image types are expected to co-exist for a very long time (this was the reason for not calling it "version").

**Comment by TheAssassin** [comment]
I do consider those legacy, they shouldn't be used any more. That doesn't mean they must not be supported. And I don't see _any_ reason to build type 1 AppImages any more. There's not a single feature missing in type 2. We even have a KDE thumbnailer now.

**Comment by probonopd** [comment]
Type 1 AppImages can be built using any(!) tool that can produce ISO9660, plus something like `dd`. This is available on practically every operating system (most likely including MS-DOS).

**Comment by TheAssassin** [comment]
`mksquashfs` is platform independent, too. Also, one can just `cat <runtime from GitHub> <squashfs image> > AppImage`, too. That's nothing type 1 could do only.
